### DISCUSSION TITLE
The method dataFetcher(String, DataFetcher) from the type TypeRuntimeWiring.Builder refers to the missing type String

### DISCUSSION THREAD

**Discussion by LucianViana** [graphql-java/graphql-java#3500] [General]
I couldn't find an example that can use graphql-java in a microservice other than spring boot?

Can't I use the graphql-java "Getting start" example just by including the dependency in any maven project?
or does it need to be a Spring project with Handler etc?

https://www.graphql-java.com/documentation/getting-started/
Error:
"The method dataFetcher(String, DataFetcher) from the type TypeRuntimeWiring.Builder refers to the missing type String"

**Comment by bbakerman** [comment]
You dont have to use Spring - you can wire together code yourself

Are you saying that hello world is not working for you - I can run it successfuly.

If you dont use spring then you need to do the conversion between the wire protocol (JSON say) and the objects the graphql engine expects.

But plenty of people do that outside Spring.  I urge you to read the doco pages for the various aspects of graphql-java

there is an old school Jetty example here

https://github.com/graphql-java/graphql-java/blob/master/src/test/groovy/example/http/HttpMain.java

**Comment by bhabegger** [comment]
FYI: I wrote and introduction article where there is only graphql-java as a dependency (and unit test stuff) showing how to use GraphQL independently of anything:

https://medium.com/@benjamin.habegger/step-by-step-guide-to-build-a-graphql-server-in-java-over-multiple-data-sources-30c0c7b4b409

Thought this might help ;)

**Reply by bbakerman** [reply]
that tutorial is great!

**Reply by dondonz** [reply]
Thanks for writing such a detailed tutorial! We were thinking of having a page to highlight tutorials from the community, would you be happy for us to link to your page on the documentation?

**Reply by bhabegger** [reply]
Yes, please do :)

**Reply by dondonz** [reply]
Thanks I've added it to the tutorial page on the documentation. The link will appear in a community tutorial section on this page https://www.graphql-java.com/tutorials/getting-started-with-spring-boot in a few minutes.

I copied your name as it appears on your Medium tutorial and I linked to your GitHub profile, but if you would like to be credited in a different way, I'm happy to change it.

PR: https://github.com/graphql-java/graphql-java-page/pull/150

**Reply by bhabegger** [reply]
Thanks @dondonz  !

**Reply by bhabegger** [reply]
@dondonz  FYI : I've gone to the next step, a Udemy course is now available : https://www.udemy.com/course/learn-graphql-in-java

**Reply by bhabegger** [reply]
GraphQL java without Spring's attached ;)

**Comment by LucianViana** [comment]
I used your example and it was incredible, thank you very much, I'm just trying to go a little further and create the fields dynamically in the "fetch" with the "Map.class" of the "runtime.execute('"countrys', Map.class)" , similar to what you said when you tried displaying the fields according to the user's hierarchy. But my question is whether I can use Map to pass dynamic fields without these fields needing to exist in the concrete class in Java or do I need to create a Map with dynamic fields like "colum_name" and "colum_value" in the concrete class and if this is the case I won't be able to search in these fields, do you agree?
Note:If it were possible to use Map to search for schema fields, it would only be necessary to change the graphql schema with new fields without having to deploy java, I believe I will have to change "graphql-java" the behavior to read dynamic data, as I already have the map<hashtag> in place of the concrete class fields, but I have no idea where to start.

**Reply by bhabegger** [reply]
@LucianViana I'm not sure I quite understand what you are trying to achieve and mean by create fields dynamically. I think you might be confusing 2 different times in the graphql workflow: schema definition time and data fetching time.

If do get the feeling you mean dynamically "defining" a field in the schema, based on some meta data from an external source (for example the schema of a database). Am I right ? 

Maybe you can explain a little bit more concretely with an example ?

**Reply by LucianViana** [reply]
Thank you, I already bought it and now I'm going to study it, I didn't do the example in git because I couldn't return it as a map from fetch as it says in the graphql-java documentation and I was only able to return it as a type with fields defined in the schema.
Thanks
